process.go

```diff
@@ -24,48 +24,34 @@ func (v *Venom) Process(path []string, exclude []string) (*Tests, error) {
 	log.SetOutput(v.LogOutput)
 	switch v.OutputDetails {
 	case DetailsLow, DetailsMedium, DetailsHigh:
-		log.Infof("Detail Level: %s", v.OutputDetails)
+		log.Debug("Detail Level: ", v.OutputDetails)
 	default:
-		return nil, errors.New("Invalid details. Must be low, medium or high")
-	}
-
-	filesPath, err := getFilesPath(path, exclude)
-	if err != nil {
-		return nil, err
+		return errors.New("Invalid details. Must be low, medium or high")
 	}
 
-	if err := v.readFiles(filesPath); err != nil {
-		return nil, err
-	}
+	return nil
+}
 
-	//First parse the testsuites to check if all testsuites are fine (context init + variable usages)
-	if err := v.parse(); err != nil {
-		return nil, err
+// Parse parses tests suite to check context and variables
+func (v *Venom) Parse(path []string, exclude []string) error {
+	if err := v.init(); err != nil {
+		return err
 	}
 
-	//Then process
-	if v.OutputDetails != DetailsLow {
-		pool := v.initBars()
-		defer endBars(v.OutputDetails, pool)
+	filesPath, err := getFilesPath(path, exclude)
+	if err != nil {
+		return err
 	}
 
-	for i := range v.testsuites {
-		ts := &v.testsuites[i]
-		v.runTestSuite(ts)
+	if err := v.readFiles(filesPath); err != nil {
+		return err
 	}
 
-	testsResult := &Tests{}
-	v.computeStats(testsResult)
-
-	return testsResult, nil
-}
-
-func (v *Venom) parse() error {
 	missingVars := []string{}
 	extractedVars := []string{}
 	for i := range v.testsuites {
 		ts := &v.testsuites[i]
-		log.Info("Parsing testsuite %s", ts.Package)
+		log.Info("Parsing testsuite", ts.Package)
```

Comment:
log.Infof("Parsing testsuite %s", ts.Package)

Reply:
when the `%s` is at the end it's no very usefull in fact
https://play.golang.org/p/aUQ6D9vKy2